chrome/content/zotero/xpcom/reader.js

```diff
@@ -1831,6 +1830,8 @@ class Reader {
 				}
 			});
 			this._readers.push(reader);
+			// Change tab's type from "unloaded-reader" to "reader" after reader loaded
+			win.Zotero_Tabs.markAsLoaded(tabID);
```

Comment:
On windows (especially slower ones like my VM), this may lead to a kind of race condition. When tab A is loading, if we click from it onto another tab B and then click back into the original tab A, then (if reader has not finished loading) as far as [Zotero_Tabs.select](https://github.com/zotero/zotero/blob/5a6b1a9f76dd42b09cb820907aa0cd6761cf641f/chrome/content/zotero/tabs.js#L454) is concerned, tab A is still `reader-unloaded` so it will call `Zotero.Reader.open` again. This can lead to multiple reader instances being opened in the same tab. That was the main reason behind placing the `Zotero_Tabs.markAsLoaded` at the top of the function (which 100% needed to be explained in a comment)

This is what it would look like if I start the app and, while things are loading, click between two tabs a few times:

<img width="1328" alt="Screenshot 2024-06-27 at 9 47 27 AM" src="https://github.com/zotero/zotero/assets/36271954/442b0358-7a00-418b-8927-bbf1ba8b7f1a">

Something I thought of earlier was to introduce another tab type, like `reader-loading`, to set this status at the top of the function and change it to just `reader` when everything is loaded. Not sure if it would lead to conflicts with anything else though

Reply:
Might it be from using `Zotero_Tabs.selectedID`? We could use the `extraData[ids[0]].type` from the notifier callback arguments, which will be called only once for each tab.

Reply:
I'm setting the tab to `reader-loading` immediately. Hopefully, that gets rid of the duplication loading issue.